h2o-core/src/main/java/water/util/VecUtils.java

```diff
@@ -838,4 +838,66 @@ public void map(Chunk c, NewChunk nc) {
     }
   }
 
+  /**
+   * shuffle values of a Vec {@link Vec}.
+   * 
+   * The 'new' values of the vector will be the already exisitng one's however randomly shuffeled
+   * 
+   * 
+   */
+  
+  public static class shuffleVec extends MRTask<shuffleVec>{
+    public Vec _vec;
+        
+    public shuffleVec(Vec vec) {this._vec = vec;}
+        
+        @Override
+        public void map(Chunk ch/*, NewChunk nc*/) {
+//          nc.alloc_doubles(ch._len);
+                
+            for (int i = 0; i < ch._len; ++i) {
+                int randIndex = i + (int) (Math.random() * (ch._len - i));
+                double vecVal = ch.atd(i);
+                double nextvecVal = ch.atd(randIndex);
+                _vec.set(i, nextvecVal);      // this
+                _vec.set(randIndex, vecVal);  // and this lines must be expensive! *For now leaving it like this
+            }
+        }
+            
+        @Override 
+        public void reduce(shuffleVec mrt) {
+            if( _vec != mrt._vec ) _vec = mrt._vec;
```

Comment:
3. In the reduce phase, you always replace the whole changed copy of `_vec` from one node to another changed copy of `_vec` from another node. When you try to run this code on big data, it will return you a dataset, where the first `ch._len` of rows will be shuffled, but other rows will be in the same order.